### PR TITLE
Fill the Z Position in UI

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/3d-tasks/issues/9115

### Changelog

* Remove fill hack on sprite simple mode
* Remove fill hack on label ttf mode
* Fill z position in all ui

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check // Manual trigger with `@cocos-robot run test cases` afterward

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/2d/assembler/label/ttf.ts [hunk 1]
```diff
@@ -46,7 +47,7 @@ export const ttf: IAssembler = {
     createData (comp: Label) {
         const renderData = comp.requestRenderData()!;
 
-        renderData.dataLength = 2;
```
Comment (SantyWang) [inline cocos/2d/assembler/label/ttf.ts:49]: why change to 4

File: cocos/2d/assembler/sprite/bar-filled.ts [hunk 2]
```diff
@@ -222,9 +222,11 @@ export const barFilled: IAssembler = {
         const vData = chunk.vb;
 
         let offset = 0;
+        const z = node.getPosition().z;
```
Comment (SantyWang) [inline cocos/2d/assembler/sprite/bar-filled.ts]: useless code?

File: cocos/2d/assembler/label/ttf.ts (hunk 1 above)
Comment (LinYunMo) [reply cocos/2d/assembler/label/ttf.ts:49]: same as other assembler

**Review by SantyWang — APPROVED** [review]

**Comment by SantyWang** [comment]
@cocos-robot run test cases

**Review by pandamicro — APPROVED** [review]

**Pull request merged**